packages/dds/tree/src/feature-libraries/forest-proxy/forestProxy.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { FieldKey } from "../../tree";
+import { IEditableForest, TreeNavigationResult, ITreeSubscriptionCursor } from "../../forest";
+
+export const proxySymbol = Symbol("forest-proxy");
+
+class TargetForest {
+	private readonly _cursor: ITreeSubscriptionCursor;
+
+	constructor(
+		public readonly forest: IEditableForest,
+		_cursor?: ITreeSubscriptionCursor,
+	) {
+		if (_cursor) {
+			this._cursor = _cursor.fork();
+		} else {
+			this._cursor = forest.allocateCursor();
+			forest.tryMoveCursorTo(forest.root(forest.rootField), this._cursor);
+		}
+	}
+
+	public get cursor(): ITreeSubscriptionCursor {
+		return this._cursor;
+	}
+
+	public get type() {
+		return this.cursor.type;
+	}
+}
+
+const handler: ProxyHandler<TargetForest> = {
+	get: (target: TargetForest, key: string): any => {
+		const result = target.cursor.down(key as FieldKey, 0);
+		if (result === TreeNavigationResult.NotFound) {
+			return Reflect.get(target, key);
```

Comment:
When the current node lacks the requested key, why is the fallback behavior to reflect on the TargetForest?

Is this perhaps working by coincidence since the Reflect.get(..) will typically return 'undefined' (unless you ask for 'cursor', 'type', etc.)

Reply:
> When the current node lacks the requested key, why is the fallback behavior to reflect on the TargetForest?
> 
> Is this perhaps working by coincidence since the Reflect.get(..) will typically return 'undefined' (unless you ask for 'cursor', 'type', etc.)

Not really in this place. Currently it is not needed, you are right, but there are ideas to maybe enhance the target with some utilities so that this pattern will provide an access to them.